### ISSUE TITLE
Support inline code spans inside emphasis

### ISSUE THREAD

**Issue by davidsierradz** [vimwiki/vimwiki#709]
I think conceal is broken when using backticks inside bold or italic sentences:

Input | Spec | Vimwiki rendering
-- | -- | --
\*\*hello \`world\`\*\*| **hello `world`** | \*\*hello `world`\*\*
\*hello \`world\`\*| *hello `world`* | \*hello `world`\*

init.vim:

```viml
let g:vimwiki_list = [
      \ {'path': '~/notes/',
      \ 'links_space_char': '-',
      \ 'syntax': 'markdown', 'ext': '.md'}]

let g:vimwiki_global_ext = 0

let g:vimwiki_folding = 'custom'

let g:vimwiki_auto_header = 1

nnoremap <F4> "=strftime("%Y-%m-%d_%H:%M:%S")<CR>P
inoremap <F4> <C-R>=strftime("%Y-%m-%d_%H:%M:%S")<CR>
iab <expr> dts strftime("%Y-%m-%d_%H:%M:%S")

autocmd FileType vimwiki setlocal fdm=marker fmr={{{,}}}
```

```
:VimwikiShowVersion

Branch: dev

Revision: cf8a16c

Date: 2019-05-11 14:01:45 -0600
```

**Comment by ranebrown** [comment]
See [commonmark spec](https://spec.commonmark.org/0.29/#emphasis-and-strong-emphasis) rule 17 and [example 477](https://spec.commonmark.org/0.29/#example-477).

**Comment by tinmarino** [comment]
Also the "inline" code must support multiple lines as [example 635](https://spec.commonmark.org/0.29/#example-637) and issue #847

**Comment by tinmarino** [comment]
I'm happy to announce that this has been fixed by the last referencing commit: 94a7885

## Commit Description

* It uses regions syntaxes instead of previous matches:
https://github.com/vimwiki/vimwiki/blob/94a78859d7200118afa8f6e36c2f5315476c683e/autoload/vimwiki/u.vim#L244-L261

* The magic of the nesting is from these lines: https://github.com/vimwiki/vimwiki/blob/94a78859d7200118afa8f6e36c2f5315476c683e/autoload/vimwiki/u.vim#L279-L283

* It includes extended tests: https://github.com/vimwiki/vimwiki/blob/94a78859d7200118afa8f6e36c2f5315476c683e/test/syntax.vader#L27-L42

## Screenshot

So at the end it looks like this (conceal=3, conceal=0)

![Screenshot from 2020-08-03 18-50-44-vimwiki-typeface](https://user-images.githubusercontent.com/6123962/89234903-0f44f700-d5bb-11ea-871c-09ea9cf087d3.png)

### Get strikethrough  working (gnome-terminal, tmux)

To get the new strikethrough font, you need vim >= 8.1 and a terminfo well configured and not disconfigured by tmux or screen

### Bashrc

```bash
if command -v tmux &> /dev/null && [ -n "$PS1" ] \
    && [[ ! "$TERM" =~ screen ]]  && [[ ! "$TERM" =~ "screen-256color" ]] \
    && [[ ! "$TERM" =~ tmux ]] && [[ ! "$TERM" =~ "tmux-256color" ]] \
    && [ -z "$TMUX" ]; then
  exec env TERM=tmux-256color tmux
fi
```

### TmuxRc

```
# get strikethought
set -as terminal-overrides ',*:smxx=\E[9m'
set -g default-terminal "tmux-256color"
```

Thanks for the issue with such a nice array.
